rq/worker.py

```diff
@@ -698,6 +698,7 @@ def perform_job(self, job, queue):
         try:
             with self.death_penalty_class(job.timeout or self.queue_class.DEFAULT_TIMEOUT):
                 rv = job.perform()
+            job.refresh()
```

Comment:
After thinking about this a bit further, I have a better solution. Instead of reading `meta` from Redis and saving it (race condition could also happen here), we should modify the `save()` call in `handle_job_success()` to exclude `meta` so the job key in Redis will keep its `meta` value.

Something like this would be great: `job.save(include_meta=False)`.